common/helpers.go

```diff
@@ -308,12 +309,37 @@ func waitForEvent(ctx context.Context, emitter EventEmitter, events []string, pr
 	return nil, nil
 }
 
-// k6Throw throws a k6 error
+// k6Throw throws a k6 error, and before throwing the error, it finds the
+// browser process from the context and kills it if it still exists.
+// TODO: test
 func k6Throw(ctx context.Context, format string, a ...interface{}) {
 	rt := k6common.GetRuntime(ctx)
 	if rt == nil {
 		// this should never happen unless a programmer error
 		panic("cannot get k6 runtime")
 	}
-	k6common.Throw(rt, fmt.Errorf(format, a...))
+	defer k6common.Throw(rt, fmt.Errorf(format, a...))
+
+	pid := GetProcessID(ctx)
+	if pid == 0 {
+		// this should never happen unless a programmer error
+		panic("cannot find process id")
+	}
+	p, err := os.FindProcess(pid)
+	if err != nil {
+		// optimistically return and don't kill the process
+		return
+	}
+	ps, _ := p.Wait()
+	if err != nil {
```

Comment:
Did you mean to assign to `err` from the `Wait()` call? Otherwise you're just checking the above `err`.

And I'm not sure about the flow here. Wouldn't this wait for the browser to finish exiting normally, and what we want is to kill it prematurely?

Reply:
Ah, good catch, thanks. 

I thought we can gracefully shut down our extension but yeah, we better kill it prematurely.